plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineStandardDeployCommandHandler.java

```diff
@@ -3,17 +3,94 @@
 import org.eclipse.core.commands.AbstractHandler;
 import org.eclipse.core.commands.ExecutionEvent;
 import org.eclipse.core.commands.ExecutionException;
-import org.eclipse.jface.dialogs.MessageDialog;
-import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.ISources;
+import org.eclipse.wst.common.project.facet.core.IFacetedProject;
 
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Command handler to deploy an App Engine web application project to App Engine Standard.
+ * <p>
+ * It copies the project's exploded WAR to a staging directory and then executes staging and deploy operations
+ * provided by the App Engine Plugins Core Library.
+ */
 public class AppEngineStandardDeployCommandHandler extends AbstractHandler {
 
+  private ProjectToStagingExporter projectToStagingExporter;
+  private StagingDirectoryProvider stagingDirectoryProvider;
+  private FacetedProjectHelper facetedProjectHelper;
+  
+  public AppEngineStandardDeployCommandHandler() {
+    this(new ProjectToStagingExporter(), new DialogStagingDirectoryProvider(), new FacetedProjectHelper());
+  }
+  
+  @VisibleForTesting
+  AppEngineStandardDeployCommandHandler(ProjectToStagingExporter exporter, 
+                                        StagingDirectoryProvider stagingDirectoryProvider,
+                                        FacetedProjectHelper facetedProjectHelper) {
+    if (exporter == null) {
```

Comment:
I just removed the null checks on the assumption that this method will only be used in tests and if they are not set, then we'll have test failures, but not production code would break.